src/api/c/blas.cpp

```diff
@@ -27,10 +27,12 @@ static inline af_array sparseMatmul(const af_array lhs, const af_array rhs,
 }
 
 template<typename T>
-static inline af_array matmul(const af_array lhs, const af_array rhs,
-                              af_mat_prop optLhs, af_mat_prop optRhs) {
-    return getHandle(
-        detail::matmul<T>(getArray<T>(lhs), getArray<T>(rhs), optLhs, optRhs));
+static inline void matmul(af_array *out,
+                          const af_array lhs, const af_array rhs,
+                          af_mat_prop optLhs, af_mat_prop optRhs) {
+    detail::matmul<T>(getArray<T>(*out),
```

Comment:
...
This assumes `out` is already created. What happens if someone wants to create out instead of reusing an existing one ?

Reply:
I think your comment below (quoting `src/api/c/blas.cpp:150`) answers this question. I'll respond to that comment shortly